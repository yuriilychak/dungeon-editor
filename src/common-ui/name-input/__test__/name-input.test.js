import React from "react";
import {createMount} from "@material-ui/core/test-utils";

import NameInput from "../name-input";

describe( "NameInput test", () => {

    const props = {
        isError: false,
        titleLocale: "test locale",
        errorLocale: "test error",
        maxLength: 20
    };

    const wrapper = createMount()(
        <NameInput
            {...props}
        />
    );
    const instance = wrapper.instance();

    it("Default snapshot", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it("Error snapshot", () => {
        const errorProps = {...props, isError: true};
        const errorWrapper = createMount()(
            <NameInput
                {...errorProps}
            />
        );

        expect(errorWrapper.html()).toMatchSnapshot();
    });

    it("Test input value: not empty", () => {
        instance.onProjectNameChange({
            target: {
                value: "test_text difference"
            }
        });

        expect(instance.state.projectName).toBe("test_text");
    });

    it("Test input value: empty", () => {
        instance.onProjectNameChange({
            target: {
                value: " "
            }
        });

        expect(instance.state.projectName).toBe("");
    });
});
