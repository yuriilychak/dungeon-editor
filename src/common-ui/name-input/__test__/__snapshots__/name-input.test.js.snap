// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameInput test Default snapshot 1`] = `"<div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\">test locale</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" autocomplete=\\"off\\" type=\\"text\\" maxlength=\\"20\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div></div>"`;

exports[`NameInput test Error snapshot 1`] = `"<div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-error Mui-error\\" data-shrink=\\"false\\">test error</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline Mui-error Mui-error MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"true\\" autocomplete=\\"off\\" type=\\"text\\" maxlength=\\"20\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div></div>"`;
